app/processors/casino/processor_concern/ticket_granting_tickets.rb

```diff
@@ -60,6 +67,22 @@ def cleanup_expired_ticket_granting_tickets(user)
         CASino::TicketGrantingTicket.cleanup(user)
       end
 
+      def generate_two_factor_authenticator(user, processor)
+        return if processor.listener.respond_to?(:disable_two_factor_auth?) && processor.listener.disable_two_factor_auth?
+
+        two_fa_listener  = CASino::TwoFactorAuthenticatorRegistratorListener.new(processor.listener.controller)
+        two_fa_processor = CASino::TwoFactorAuthenticatorRegistratorProcessor.new(two_fa_listener)
+        two_fa_processor.process(user)
+      end
+
+      def generate_otp(ticket_granting_ticket)
+        authenticator = ticket_granting_ticket.user.two_factor_authenticator
+        totp = ROTP::TOTP.new(authenticator.secret, interval: CASino.config.two_factor_authenticator[:lifetime])
+
+        otp_value = totp.now
+        login_otp_proxy = Proxies::LoginOTP.new(otp: otp_value, person_id: ticket_granting_ticket.user.extra_attributes[:person_id])
+        Notifikator.generate :login_2fa_auth, login_otp: login_otp_proxy
```

Comment:
do we use our Notifikator gem here? i couldn't see it in the Gemfile or Gem specs.

Reply:
it is fetched from Reaktor